src/App.tsx

```diff
@@ -1,63 +1,111 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
+
 import 'bulma/css/bulma.css';
 import './App.scss';
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+enum SortType {
+  NAME = 'name',
+  LENGTH = 'length',
+}
+
+const goodsFromServer = [
+  { id: 1, name: 'Dumplings' },
+  { id: 2, name: 'Carrot' },
+  { id: 3, name: 'Eggs' },
+  { id: 4, name: 'Ice cream' },
+  { id: 5, name: 'Apple' },
+  { id: 6, name: 'Bread' },
+  { id: 7, name: 'Fish' },
+  { id: 8, name: 'Honey' },
+  { id: 9, name: 'Jam' },
+  { id: 10, name: 'Garlic' },
 ];
 
-export const App: React.FC = () => {
+function getPreparedGoods(goods: { id: number; name: string }[],
+  { sortField, reverseList }:
+  { sortField: SortType | ''; reverseList: boolean }) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.NAME:
+          return good1.name.localeCompare(good2.name);
+        case SortType.LENGTH:
+          return good1.name.length - good2.name.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseList) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { sortField, reverseList: isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SortType.NAME })}
+          onClick={() => setSortField(SortType.NAME)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SortType.LENGTH })}
+          onClick={() => setSortField(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
it's better not to set getter into setter. Plus move this logic to a function
Example: 
```
const handleReverseClick = () => {
  setIsReversed(prev => !prev)
}
```
```suggestion
          onClick={handleReverseClick}
```